src/test/java/net/openhft/chronicle/map/ChronicleMapTest.java

```diff
@@ -1881,6 +1882,52 @@ public void testBytesMarshallableMustBeConcreteValueType() {
         }
     }
 
+    @Ignore //I've been using this class as test harness to probe the map configuration and how it re-sizes.
+    @Test
+    public void testFillUpMapWhilstDemonstratingMonitor() throws IOException {
+        char[] chars = new char[2 << 9]; //Writing 512 char entries
+        Arrays.fill(chars, 'A');
+        String val = new String(chars);
+        int entries = 200_000;
+
+        try (final ChronicleMap<Integer, String> map = ChronicleMapBuilder
+                .of(Integer.class, String.class)
+                //.averageKey(8) Not required as we are using Integer keys and size is implicit
+                .averageValue(val)
+                .entries(entries)
+                .maxBloatFactor(1)
+                .minSegments(1).create()) {
+            reportMapUsage(map);
+
+            int LOW = entries;
+            int MED = (2 * entries);
+            int HIGH = (3 * entries);
+            int ABOVE_CAPACITY = (4 * entries);
+
+            MapCapacityMonitor monitor = new MapCapacityMonitor(map, 10); //10% for visibility
+            for (int i = 1; i < HIGH; i++) { //Change to LOW / MED / HIGH / ABOVE_CAPACITY to see the monitor warnings
+                map.put(i, val);
+                assertEquals(i, map.size());
+                if (i % 20000 == 0) {
+                    reportMapUsage(map);
+                }
+            }
+        }
+    }
+
+    private static void reportMapUsage(ChronicleMap<? extends Object, ? extends Object> map) {
```

Comment:
I added this so I can see how the metrics provided by Map can change as it is under stress.